src/java.base/share/classes/java/util/concurrent/FutureTask.java

```diff
@@ -205,6 +205,59 @@ public V get(long timeout, TimeUnit unit)
         return report(s);
     }
 
+    @Override
+    public V resultNow() {
+        switch (state()) {    // Future.State
+            case SUCCESS:
+                @SuppressWarnings("unchecked")
+                V result = (V) outcome;
+                return result;
+            case FAILED:
+                throw new IllegalStateException("Task completed with exception");
+            case CANCELLED:
+                throw new IllegalStateException("Task was cancelled");
+            default:
+                throw new IllegalStateException("Task has not completed");
+        }
+    }
```

Comment:
I think the code will be more readable if a switch expression and the arrow syntax are used

```
return switch(state()) {
    case SUCCESS -> {
        @SuppressWarnings("unchecked")
        V result = (V) outcome;
        yield result;
  } 
  case FAILED -> throw new IllegalStateException("Task completed with exception");
 ...
};
```

Reply:
Sorry, I don't understand how it would be more readable. I like new switch features because they extend the range of applicability of switch. But this (and the two others) are just plain vanilla enum-based switches that were handled well, and seem to have the same numbers of lines and structure either way?

Reply:
It's more readable because it's now clear that the aim is to return the value of the switch.
Also a switch expression has to be exhaustive, so there is no need for a 'default'.